core/src/application_restarter.cpp

```diff
@@ -43,6 +51,6 @@ int ApplicationRestarter::restart(int exitCode)
 
 void ApplicationRestarter::triggerRestart()
 {
-	qApp->setProperty("restart", QVariant(true));
+	m_restart = true;
```

Comment:
You can make this method static by getting the GetInstance() inside the method. Makes it easier to call ..